### DISCUSSION TITLE
Thoughts on Extensions

### DISCUSSION THREAD

**Discussion by halr9000** [anapnoe/stable-diffusion-webui-ux#73] [Ideas]
Starting a quick discussion that might lead to enhancement requests but my thoughts aren't all the way in order for that yet. 

# What works

- Lists available extensions, providing good info to help you to decide whether or not to install at a glance
- [Gitops approach](https://www.redhat.com/en/topics/devops/what-is-gitops) leads to best practices
- The list itself is dynamic, stored as [a file](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/blob/master/index.json) in [a repo](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions)
  - This repo actually looks pretty healthy! Many active contributors.
- Installation is quick and painless
- Updating is also quick and easy

# What doesn't work / is missing

- Can't manage full lifecycle (CRUD, using a REST API analogy): missing ability to modify (U) or remove (D)
- Updates have no context, you know know you are "behind"
- UI is pretty minimal, hasn't had any attention. Lots of little changes could improve UX.
  - ordering. remove "internal order" sort, nobody cares, can optimize space using conventional table/column header type controls
  - too many tags to list them all, probably needs a combo box approach
  - inverse of hide tags is what I want sometimes, i.e. "show me themes only"
  - horizontal space is wasted, can add some columns into the table like dates (added/updated)
- Extension list is pretty binary: either an extension is "production" and in the list, or it doesn't exist and nobody knows to look for it and experimental stuff won't get many eyeballs / little community effect.
  - Some more thinking needed here, but plenty of ideas...add a concept of experimental, beta, stable. 
  - Probably need to make a new discussion over in the extensions repo.

**Comment by anapnoe** [comment]
these are all good points and will be addressed separately in the future starting from the
1. UI can be improved agree completely on this
2. versioning system for ui and extensions are very important there is none at the moment we just use commits to ref this has create compatibility nightmares 
3. REST API needs a lot of work as you said to support the full lifecycle but this can wait and need more brainstorming on how to also work for extensions

**Reply by halr9000** [reply]
> compatibility nightmares
Ugggh, having Gradio flashbacks don't remind me lol
